### DISCUSSION TITLE
Check Syntax?

### DISCUSSION THREAD

**Discussion by DustinDunbar** [vscode-abl/vscode-abl#152] [Q&A] [answered]
Stupid starter question... been using the predecessor for this plugin for years.  Shift-F2 was always check syntax:
also set the procedure editor key bindings.
![image](https://github.com/vscode-abl/vscode-abl/assets/151785828/1e00b0c7-a48f-4c21-a0e2-e6580dee562f)

Is this supposed to trigger a syntax check?  Or am I missing something fundamental - like what I should be looking at?

**Comment by PeterJudgeZA** [comment]
AFAIK, there is no "check syntax" functionality for programs in a project. The language server tries to compile on each save and marks the explorer and editor with any issues (red colour, squigglies on the lines). 

I assume that this would work just as well with VSCode's auto-save functionality

It's an adjustment from PDSOE (or the procedure editor) but you get used to it eventually :D

There's a "compile current buffer" tool but 'm not sure whether it's intended as a 'check syntax' alternative.

**Comment by DustinDunbar** [comment]
Thanks Peter, that is more or less my question, the problem right now is I'm not seeing any squiggles? That would work for me!

It was working under the previous plugin.

**Reply by PeterJudgeZA** [reply]
It seems to me that sometimes the language server gets confused ... the 'tech support 101' approach I take is
1. Close VSCode
2. Open  the project(s) in a terminal/UI.
3. Delete the entire .builder folder(s)
4. Restart VSCode

It's a bit extreme, but this usually results in things behaving again.

**Reply by DustinDunbar** [reply, marked as answer]
Ahh. took me a bit more but realized I had to add the directories to the buildPath as source. I had setup the propath entries.  With the addition of source the squiggles started appearing

Thanks Again!

**Comment by gquerret** [comment]
"Compile current buffer" is meant to have the same functionality as "Check syntax". Action is not bound to Shift F2 by default, but users can assign keybindings to actions. This action is useful when working on unsaved buffers, otherwise I think it's faster to use Ctlr + S.

**Reply by DustinDunbar** [reply]
That makes sense, I'll probably remap that. 

Any reason why f2 has to run on a saved program/file then? it seems like F2 should run the current buffer - but I see it maintains the behaviour of the previous plugin - that is only running the saved program

**Reply by gquerret** [reply]
VSCode is **not** a replacement for procedure editor, so it can't do the quick and dirty things that you can do with the procedure editor. It's a project oriented development environment, much more similar to PDSOE than procedure editor.
